### PR TITLE
Add lambda defaults/`params` feature

### PR DESCRIPTION
**Pull request by jjonescz**
Relates to test plan #62485.

- I've successfully built runtime locally with this version of Roslyn.
- [VS insertion](https://dev.azure.com/devdiv/DevDiv/_git/VS/pullrequest/436006) fails while building the installer with the same errors as previous insertions, so nothing related to this feature.

### CODE REVIEW

File: src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/LambdaConversionResult.cs [hunk 1]
```diff
@@ -12,6 +12,7 @@ internal enum LambdaConversionResult
         MissingSignatureWithOutParameter,
         MismatchedReturnType,
         MismatchedParameterType,
+        MismatchedParameterDefaultValue,
```
Comment (jcouv) [inline src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/LambdaConversionResult.cs:15]: Is this value (`MismatchedParameterDefaultValue`) still used?

File: src/Compilers/Core/Portable/Collections/Rope.cs [hunk 2]
```diff
@@ -86,6 +88,23 @@ private sealed class StringRope : Rope
             private readonly string _value;
             public StringRope(string value) => _value = value;
             public override string ToString() => _value;
+
+            public override string ToString(int maxLength)
```
Comment (jcouv) [inline src/Compilers/Core/Portable/Collections/Rope.cs:92]: I think it's a GitHub bug with displaying complex diff between two branches. I don't think you'd changed this file, right?
Consider pinging GitHub support and filing a bug.

**Review by jcouv — APPROVED** [review]
LGTM Thanks (iteration 64)

**Comment by jcouv** [comment]
@jaredpar for approval to merge.

File: src/Compilers/Core/Portable/Collections/Rope.cs (hunk 2 above)
Comment (cston) [reply src/Compilers/Core/Portable/Collections/Rope.cs:92]: This change is from https://github.com/dotnet/roslyn/pull/63448.

Comment (jcouv) [reply src/Compilers/Core/Portable/Collections/Rope.cs:92]: Ah, makes sense. Thanks

**Review by jaredpar — APPROVED** [review]

**Pull request merged**